src/LCT.PackageIdentifier/MavenProcessor.cs

```diff
@@ -67,6 +84,100 @@ public Bom ParsePackageFile(CommonAppSettings appSettings)
             return bom;
         }
 
+        public static void DevDependencyIdentificationLogic(List<Component> componentsForBOM, List<Component> componentsToBOM, ref List<Component> ListOfComponents)
+        {
+                       
+            List<Component> iterateBOM = componentsForBOM.Count > componentsToBOM.Count ? componentsForBOM : componentsToBOM;
+            List<Component> checkBOM = componentsForBOM.Count < componentsToBOM.Count ? componentsForBOM : componentsToBOM;
+
+      
+            ListOfComponents = DevdependencyIdentification(ListOfComponents, iterateBOM, checkBOM);
+
+        }
+
+        private static List<Component> DevdependencyIdentification(List<Component> ListOfComponents, List<Component> iterateBOM, List<Component> checkBOM)
+        {
+            foreach (var item in iterateBOM)
+            {
+                if (checkBOM.Count == 0)
+                {
+                    SetPropertiesforBOM(ref ListOfComponents, item, "false");
```

Comment:
You can write Comments for what's happening here.

Reply:
added